src/psyclone/dynamo0p3.py

```diff
@@ -7114,6 +7114,36 @@ def __init__(self, loop_type="", **kwargs):
         self._upper_bound_name = None
         self._upper_bound_halo_depth = None
 
+    def lower_to_language_level(self):
```

Comment:
I can't quite remember the details of what this is fixing. We should probably have a test that shows what was wrong before you implemented this method? I've had a read of the associated issue and see that it was related to calling `gen` twice. Perhaps we need a test where we do this and check the output doesn't change?